scalafix-interfaces/src/main/java/scalafix/interfaces/ScalafixArguments.java

```diff
@@ -205,4 +205,14 @@ ScalafixArguments withToolClasspath(
      */
     ScalafixError[] run();
 
+    /**
+     *
+     * Compared to {@link #run()}, it doesn't write the diff to the requested file,
+     * but instead return the list of patches and diagnostics, which can be written to file using
+     * {@link ScalafixResult#writeResult()}}.
+     * withCallback doesn't have any effect, all diagnostics are directly stored in ScalafixResult
```

Comment:
can we fail or feed both rather than ignoring?

Reply:
So if the name is evaluate, it make sense that the Mode or callback are not called ?

Reply:
I think we should fail if `evaluate` is called on a `ScalafixArguments` that was explicitly set a mode or a callback, since they have no effect.

Reply:
Evaluate should not fail from my point of view. It's a new way to interact with scalafix, that requires fewer arguments.